jinad/api/endpoints/flow.py

```diff
@@ -0,0 +1,165 @@
+import uuid
+from ruamel.yaml import YAML
+from jina.flow import Flow
+from jina.clients import py_client
+from jina.excepts import GRPCServerError
+from typing import List, Union, Optional
+from fastapi import APIRouter, Body, Response
+
+from models.pod import PodBase, PodBaseExample, PodContainer
+from excepts import FlowYamlParseException, HTTPException
+from logger import get_logger
+
+logger = get_logger(context='flow-api')
+router = APIRouter()
+flow_dict = {}
+
+
+class FlowWrapper:
+    def __init__(self, 
+                 pods: List[PodBase] = None,
+                 yaml_spec=None):
+        if pods:
+            self.f = Flow()
+            self.pods = pods
+            self._build_with_pods()
+        
+        if yaml_spec:
+            yaml = YAML()
+            yaml.register_class(Flow)
+            try:
+                self.f = yaml.load(yaml_spec)
+            except Exception as e:
+                logger.error(f'Got error while loading from yaml {e}')
+                raise FlowYamlParseException
+        
+        self.start()
+
+    def _build_with_pods(self):
+        for pod in self.pods:
+            self.f = self.f.add(
+                name=pod.name,
+                uses=pod.uses
+            )
+        
+    def start(self):
+        self.f = self.f.__enter__()
```

Comment:
you can take a look at `ExitStack` to handle a `context_manager` properly handled. Not sure it is possible but this seems a bit hacky way of wrapping a context manager

Reply:
Yah, this is a little hacky :D 
Like we have a `Flow.start()` method, we can take the content of `Flow.__exit__` to `Flow.close()`. Instead of calling, context manager objects, we can just call `start` & `close`. The reason `ExitStack` cannot be used here is, we are entering the flow context in one `POST` request & exiting the context in another.

Reply:
Fair point, although I guess we can find a better way maybe in a different iteration